test/forms/ChoiceSpec.scala

```diff
@@ -26,20 +26,16 @@ class ChoiceSpec extends WordSpec with MustMatchers {
   "Validation defined in Choice mapping" should {
 
     "attach errors to form" when {
-      "provided with  empty input" in {
+      "provided with empty input" in {
         val form = Choice.form().bind(emptyChoiceJSON)
 
         form.hasErrors must be(true)
         form.errors.length must equal(1)
         form.errors.head.message must equal("choicePage.input.error.empty")
       }
 
-      "provided with a value not defined in AllowedChoiceValues" in {
-        val form = Choice.form().bind(incorrectChoiceJSON)
-
-        form.hasErrors must be(true)
-        form.errors.length must equal(1)
-        form.errors.head.message must equal("choicePage.input.error.incorrectValue")
+      "provided with an incorrect value" in {
+        intercept[IllegalArgumentException](Choice.form().bind(incorrectChoiceJSON))
```

Comment:
I think original version with form with errors is better rather then throwing exceptions

Reply:
Fixed